File: mail/mail/Model/Response.cs
namespace mail.Model;

public class Response
{
    protected Response(bool result, string message, object data)
    {
        this.result = result;
        this.message = message;
        this.data = data;
    }

    protected Response(object data)
    {
        result = true;
        message = "";
        this.data = data;
    }

    public Response(string message)
    {
        result = false;
        this.message = message;
        data = null;
    }

    public bool result { get; set; }
    public string message { get; set; }
    public object data { get; set; }
}

File: ai/response.py
from typing import List

from pydantic import BaseModel


class ResponseMenuItem(BaseModel):
    menuId: str
    Index: int


class ResponseMenuEmbedding(BaseModel):
    menuId: str
    index: int
    embedding: List[float]


class Response(BaseModel):
    data: object
    message: str
    result: bool


class RecommendMenuResponse(Response):
    data: List[ResponseMenuItem]


class MenuEmbeddingResponse(Response):
    data: List[ResponseMenuEmbedding]
